diff --git a/meson.build b/meson.build
index be768af..6a0950c 100644
--- a/meson.build
+++ b/meson.build
@@ -281,7 +281,7 @@ add_project_arguments('-include', 'config.h', language : 'c')
 inc_root = include_directories('.')
 inc_drm = include_directories('include/drm')
 
-libdrm = shared_library(
+libdrm = static_library(
   'drm',
   [files(
      'xf86drm.c', 'xf86drmHash.c', 'xf86drmRandom.c', 'xf86drmSL.c',
